### ISSUE TITLE
Call function that was passed as argument of java callback

### ISSUE THREAD

**Issue by ldiqual** [eclipsesource/J2V8#28]
Is this something possible with the current state of the code? I looked for `->call()` in the jni code but couldn't find any, I suppose that function calls are not supposed from the Java side?

**Comment by irbull** [comment]
It's not easily support (and it's probably a good idea to have a first class V8Function that extends V8Object), but you can do something like this:
```
    public void callback(final V8Object object) {
        V8Array parameters = new V8Array(v8);
        parameters.push((V8Object) null);
        parameters.push("John");
        parameters.push("Smith");
        parameters.push(7);
        object.executeVoidFunction("call", parameters);
        parameters.release();
    }

    @Test
    public void testVoidAnonymousFunctionCall() {
        v8.executeVoidScript("var func = function(first, last, age) {person = {'first':first, 'last':last, 'age':age};}");
        v8.registerJavaMethod(this, "callback", "callback", new Class<?>[] { V8Object.class });

        v8.executeVoidScript("callback(func);");
        V8Object result = v8.getObject("person");

        assertEquals("John", result.getString("first"));
        assertEquals("Smith", result.getString("last"));
        assertEquals(7, result.getInteger("age"));
        result.release();
    }
```

Notice we pass 'func' to the callback, and execute the 'call' function on it. In this case I just use null for 'this'. Not sure what the best value is there.

**Comment by irbull** [comment]
Fixed with 2af3d6633e6079c2f1157bc90d27b4c5d974de3b. We now have a proper V8Function with a call method on it.
